Initial Commit

Warhammer4000 opened the pull request:
Known issue
-Event subscription bug : Trying to play new dialogue before one ends ends up creating a bug that breaks the text display

Warhammer4000 committed:
Initial Commit

Known issue
-Event subscription bug : Trying to play new dialogue before one ends ends up creating a bug that breaks the text display

N-Evan commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Editor/Data/CharacterTableEditor.cs:
```diff
@@ -0,0 +1,192 @@
+        private CharacterTable characterTable;
+        private string characterID = "";
+        private string characterName = "";
+        private string csvFilePath = "";
```
Did not follow the variable naming convention.

N-Evan commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Core/DialogueManager.cs:
```diff
@@ -0,0 +1,56 @@
+            DialogueUI.RegisterGraph(_currentGraph);
+            _currentGraph.StartDialogue();
+        }
+
+        
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+    }
```
Unnecessary whitespace.

N-Evan commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Core/DialogueManager.cs:
```diff
@@ -0,0 +1,56 @@
+{
+    public class DialogueManager : MonoBehaviour
+    {
+
```
Unnecessary space.

N-Evan commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Core/DialogueUI.cs:
```diff
@@ -0,0 +1,111 @@
+    public class DialogueUI : MonoBehaviour
+    {
+        [Header("UI Components")]
+        public TextMeshProUGUI dialogueText;
```
Did not follow naming convention rules.

N-Evan commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Core/DialogueUI.cs:
```diff
@@ -0,0 +1,111 @@
+
+      
+
+        private CancellationTokenSource CancelCurrentDialogueLineToken;
```
Did not follow naming convention rules.

N-Evan commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Core/DialogueUI.cs:
```diff
@@ -0,0 +1,111 @@
+        private CancellationTokenSource CancelCurrentDialogueLineToken;
+
+
+        void Start()
```
Must explicitly mention access modifiers.

N-Evan commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Core/DialogueUI.cs:
```diff
@@ -0,0 +1,111 @@
+            _currentGraph.ClearEvents();
+            _currentGraph.OnDialogueStart += () => PlayDialogue();
+            _currentGraph.OnDialogueComplete += () => EndDialogue();
+            _currentGraph.OnDialogueNext += async () => await ShowDialogueText(_currentGraph.CurrentNode.DialogueText);
```
Lacks readability considering everyone is still not familiar with lambda expression short-hands mixed with async function calls.

N-Evan commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Data/Configuration/DialogueUIConfig.cs:
```diff
@@ -0,0 +1,21 @@
+    public class DialogueUIConfig : ScriptableObject
+    {
+        [Header("Text Display Settings")]
+        public float letterDelay = 0.05f;
+        public float nextSentenceDelayMultiplier = 1.0f;
+
+        [Header("Subtitle Configuration")]
+        public Color subtitleColor = Color.black;
+        public int subtitleFontSize = 24;
```
Did not follow naming convention rules.

N-Evan commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Data/Models/CharacterTable.cs:
```diff
@@ -0,0 +1,35 @@
+        [Serializable]
+        public class CharacterInfo
+        {
+            public string characterID;
+            public string characterName;
```
Did not follow naming convention rules.

N-Evan commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Data/Models/CharacterTable.cs:
```diff
@@ -0,0 +1,35 @@
+            public List<CharacterReactionImage> Reactions;
+        }
+
+        [Serializable]
+        public class CharacterReactionImage
+        {
+            public string Reaction;
+            public Sprite Image;
+        }
```
Let's keep every class under its separate script unless it is only used within that script and nowhere else.

N-Evan commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Data/Models/CharacterTable.cs:
```diff
@@ -0,0 +1,35 @@
+            public Sprite Image;
+        }
+
+        public List<CharacterInfo> characterList = new List<CharacterInfo>();
```
Did not follow naming convention rules.

N-Evan commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Data/xNode/DialogueBase.cs:
```diff
@@ -0,0 +1,35 @@
+﻿
+
+
+using UnityEngine;
+using XNode;
+
+
```
Extra whitespaces were not removed.

N-Evan commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Data/xNode/DialogueGraph.cs:
File contains many vertical whitespaces which are unnecessary.

N-Evan commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Data/xNode/DialogueNode.cs:
Contains unnecessary and extra vertical spaces which needs to be removed.

N-Evan commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Editor/Data/DialogueBaseNodeEditor.cs:
```diff
@@ -0,0 +1,30 @@
+    {
+
+        public Color TextColor= Color.white;
+        private DialogueBase dialogueBase;
+        private static GUIStyle editorLabelStyle;
```
Did not follow naming convention rules.

BDeshiDev commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Core/DialogueUI.cs:
```diff
@@ -0,0 +1,111 @@
+            {
+                await UniTask.Delay((int)(config.letterDelay * 1000), cancellationToken: CancelCurrentDialogueLineToken.Token).SuppressCancellationThrow();
+                if (dialogueState == DialogueState.Skipped) break;
+                dialogueText.text += text[i];
```
Can be done without string concat via textmeshpro max visible char count. Would prevent UI resizing issues as well.

N-Evan commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Core/DialogueUI.cs:
```diff
@@ -0,0 +1,111 @@
+            ShowUI(true);
+            for (int i = 0; i < text.Length; i++)
+            {
+                await UniTask.Delay((int)(config.letterDelay * 1000), cancellationToken: CancelCurrentDialogueLineToken.Token).SuppressCancellationThrow();
```
`config.letterDelay * 1000`, using arbitrary values as such creates a lack of readability in the code. Use properly named variables to ensure that other developers are able to understand the purpose. Alternatively, there's `UniTask.Delay(Timespan.FromSeconds)` and other similar methods to use in these cases.

N-Evan commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Core/DialogueUI.cs:
```diff
@@ -0,0 +1,111 @@
+
+            }
+
+            await UniTask.Delay((int)(text.Length * 100 * config.nextSentenceDelayMultiplier), cancellationToken: CancelCurrentDialogueLineToken.Token).SuppressCancellationThrow();
```
using arbitrary values as such creates a lack of readability in the code. Use properly named variables to ensure that other developers are able to understand the purpose. Alternatively, there's `UniTask.Delay(Timespan.FromSeconds)` and other similar methods to use in these cases.

N-Evan reviewed (approved):
The overall PR contains formatting and conventional issues. No major functional issues were noticed in the review so far hence approving it. 

I expect to see the commented changes in the upcoming PR and no further incorrect naming style and unnecessary whitespace in scripts. 

Edit: Additionally, please initiate the repository with the initial project files and necessary dependencies to avoid unnecessarily large PRs like this one. It would also be helpful if feature branches were named accordingly instead of a generic name like dev-x or feature. 

The PR title lacks understandability as well. With something generic as `Initial Commit` as PR title, there's no way for a reviewer or another developer to understand what was done without having to go through the code line-by-line. A bit more descriptive and functional titles are expected in the future.

BDeshiDev commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Core/DialogueUI.cs:
```diff
@@ -0,0 +1,111 @@
+namespace Studio23.SS2.DialogueSystem.UI
+{
+
+    public class DialogueUI : MonoBehaviour
```
We will need to change the UI so having a base class with overrideable PlayDialogue, EndDialogue, ShowDialogueText would be useful

BDeshiDev commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Core/DialogueManager.cs:
```diff
@@ -0,0 +1,56 @@
+        public void PlayDialogue()
+        {
+            DialogueUI.RegisterGraph(_currentGraph);
+            _currentGraph.StartDialogue();
```
Dialoguemanger should have events that fire when dialogue starts and ends so that rest of the systems can hook in without needing to know which graph initiated the dialogue.

BDeshiDev commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Data/Models/CharacterTable.cs:
```diff
@@ -0,0 +1,35 @@
+        [Serializable]
+        public class CharacterReactionImage
+        {
+            public string Reaction;
```
Assuming we want to pick reactions per dialogue node, enums for reactions would make it less error prone to select.

BDeshiDev commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Core/DialogueUI.cs:
```diff
@@ -0,0 +1,111 @@
+            BackgroundImage.color = config.BackGroundColor;
+        }
+
+
```
Extra whitespace

BDeshiDev commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Data/xNode/DialogueBase.cs:
```diff
@@ -0,0 +1,35 @@
+        public string ID;
+
+        [Header("Public Properties")]
+        public string CharacterID;
```
We could have a per character scriptableobject. Less error prone than strings

BDeshiDev commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Data/xNode/DialogueBase.cs:
```diff
@@ -0,0 +1,35 @@
+
+
+        [TextArea(3, 10)]
+        public string DialogueText;
```
Will need to replace with localization system's serialized string.

BDeshiDev commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Data/xNode/DialogueGraph.cs:
```diff
@@ -0,0 +1,84 @@
+        public DialogueEvent OnDialogueStart;
+        public DialogueEvent OnDialogueNext;
+        public DialogueEvent OnDialogueComplete;
+
```
The API should contain an event when a dialogue option is encountered so that the dialogueUI class can show it. Additionally, there should be some way to tell the graph/node that an option was picked and that it should progress.

N-Evan commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Core/DialogueUI.cs:
```diff
@@ -0,0 +1,111 @@
+            dialogueText.color = config.subtitleColor;
+            BackgroundImage.gameObject.SetActive(config.ShowBackground);
+            BackgroundImage.color = config.BackGroundColor;
+        }
```
UI-related changes and setups should be segregated to respective Ui controllers rather than having the functional script handle it.

Warhammer4000 committed:
Code update

Features Added
- An Dialogue System Wizard has been added.
- It guides the user to create character table first. Character table can be created from CSV and template can be found from the editor itself.
- Dialogue Graph can now be generated from CSV file. and it's guided from the editor wizard

Code Quality
- Renamed most variables to Adhear Guidelines.
- Removed some whitespaces

Known Issues
- Same as previous

TO-DO
-Implement Test cases
-Replace placeholder header image
- Provide samples

Warhammer4000 committed:
v0.1.9

v0.1.9
Feature List
1. Added A Dialogue System Wizard
2. It makes sure that you create a character table
3. Character table CSV template Added
4. Dialogue Graph Creator Added
5. All dialogue can now be generated via a CSV File
6. Added Dialogue Graph CSV template.

Package Changes
1. Added Samples Showing how to use the library & Some UI Code

Code Changes
1. Now the events are based on the Manager Class instead of Dialogue Graph
2. Removed Coupling from the UI Code so that new UI code can be easily rewritten

Warhammer4000 committed:
File updates

-Removed Documentation & Tests folder as it's not yet implemented
-Updated Readme file with screenshots

Warhammer4000 committed:
More file update

Updated Readme

BDeshiDev commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Editor/UITabs/DialogueGraphEditorUI.cs:
```diff
@@ -0,0 +1,163 @@
+
+        public string GraphName;
+
+        private string folderPath = $"Assets/Resources/DialogueSystem/Graphs/";
```
naming convention for private var not followed.

BDeshiDev commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Editor/UITabs/DialogueGraphEditorUI.cs:
```diff
@@ -0,0 +1,163 @@
+
+
+
+        private void PopulateDialogueGraph()
```
This populates from a csv file. It should be renamed to PopulateDialogueGraphFromCSV for clarity

BDeshiDev commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Editor/Data/DialogueBaseNodeEditor.cs:
```diff
@@ -8,7 +8,7 @@ namespace Studio23.SS2.DialogueSystem.Data
 
-        public Color TextColor= Color.white;
+        public Color TextColor = Color.white;
         private DialogueBase dialogueBase;
```
Private var naming convention not followed

BDeshiDev commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Samples/DialogueUI.cs:
```diff
@@ -0,0 +1,108 @@
+            for (int i = 0; i < text.Length; i++)
+            {
+                DialogueText.text += text[i];
+                await UniTask.Delay((int)(_config.letterDelay * 1000));
+                if (_dialogueState == DialogueState.FastForwarded)
+                {
+                    DialogueText.text = text;
+                    break;
+                }
+
+            }
+            await UniTask.Delay((int)(text.Length * 100 * _config.nextSentenceDelayMultiplier));
```
Multiplication by 1000/100 is hardcoded. To get a particular unit of Time, TimeSpan.FromSeconds and other variants can be used:

` await UniTask.Delay(TimeSpan.FromSeconds(10), ignoreTimeScale: false);`

BDeshiDev commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Editor/DialogueSystemWizard.cs:
```diff
@@ -0,0 +1,52 @@
+
+            EditorGUILayout.BeginVertical(EditorStyles.helpBox);
+
+            if (_currentTab == 0)
```
Hardcoded tab index magic numbers. Should be refactored into variables.

BDeshiDev commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Samples/DialogueUI.cs:
```diff
@@ -0,0 +1,108 @@
+            OnDialoguePlayComplete?.Invoke();
+
+        }
+
+
+
```
Unnecessary vertical space.

BDeshiDev commented on Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Data/Models/CharacterTable.cs:
```diff
@@ -24,11 +24,32 @@ public class CharacterReactionImage
+
+
+
         public List<CharacterInfo> characterList = new List<CharacterInfo>();
```
Public var naming rules not followed. Public access to this list is also unnecessary. If we want to later optimize it with a dictionary then the public list access would need to be removed.

BDeshiDev reviewed (approved):
The coupling  of the UI class with the manager is fixed. 

There are some minor changes such as incorrect var naming and  spacing issues. 

In a future version I  would recommend replacing the character table with an SO for each character info. It is possible to mess up typing CharacterID strings. With a scriptable, we can assign it and be sure that we have a correct reference without needing to write error checking code.
